### ISSUE TITLE
not able to see contiv network under "docker network ls"

### ISSUE THREAD

**Issue by shaitmm** [contiv/ansible#306]
Dear all experts,

i have setup a 1 master 2 nodes env, all of them has 2 nic(mgt+contiv_control) +nic(contiv_network).

After installation, i encountered below issue. Pls enlighten me for the solution.
If any further info is needed, just let me know.
[root@k8master ~]# netctl network ls
Tenant   Network      Nw Type  Encap type  Packet tag  Subnet       Gateway  IPv6Subnet  IPv6Gateway
------   -------      -------  ----------  ----------  -------      ------   ----------  -----------
default  default-net  data     vxlan       0           20.1.1.0/24                       
[root@k8master ~]# docker network ls
NETWORK ID          NAME                DRIVER
7d9ecb823e5f        none                null                
90a19735883c        host                host                
dc5802bd4e44        bridge              bridge 

[root@k8master k8s]# docker run -itd --net=ms --name=web alpine /bin/sh
docker: Error response from daemon: network ms not found.

Mason

**Comment by gaurav-dalvi** [comment]
@shaitmm : we already know this issue.
https://github.com/contiv/netplugin/issues/691 . We will resolve it soon !

**Comment by shaitmm** [comment]
@gaurav-dalvi so, can i just upgrade docker to 1.12.6?

below is my current docker info:
[root@k8master ~]# docker info 
Containers: 1
 Running: 1
 Paused: 0
 Stopped: 0
Images: 3
Server Version: 1.10.3
Storage Driver: devicemapper
 Pool Name: docker-253:0-201663876-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 1.526 GB
 Data Space Total: 107.4 GB
 Data Space Available: 40.48 GB
 Metadata Space Used: 2.314 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.145 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.135-RHEL7 (2016-09-28)
Execution Driver: native-0.2
Logging Driver: journald
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 3.10.0-514.2.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
Number of Docker Hooks: 2
CPUs: 4
Total Memory: 7.64 GiB
Name: k8master
ID: GAED:DBFS:FAUG:JEOP:ELGZ:MEBV:SU76:YAJ6:QRK4:EYDV:YVF3:XNB5

**Comment by shaitmm** [comment]
Dear experts,

i tried to upgrade docker on ctl node to 1.12.6 but the issue still exists.
pls enlighten me for any workaround.

[root@k8master ~]# netctl network ls
Tenant   Network      Nw Type  Encap type  Packet tag  Subnet       Gateway   IPv6Subnet  IPv6Gateway
------   -------      -------  ----------  ----------  -------      ------    ----------  -----------
default  default-net  data     vxlan       0           20.1.1.0/24                        
default  ms           data     vxlan       0           88.0.0.0/24  88.0.0.1              
[root@k8master ~]# docker network ls
NETWORK ID          NAME                DRIVER              SCOPE
3b6c73a324b5        bridge              bridge              local               
710f87111c2f        host                host                local               
47342b2ed038        none                null                local               
[root@k8master ~]# docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 1.12.6
Storage Driver: devicemapper
 Pool Name: docker-253:0-67618308-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 11.8 MB
 Data Space Total: 107.4 GB
 Data Space Available: 41.54 GB
 Metadata Space Used: 581.6 kB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.147 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.135-RHEL7 (2016-09-28)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null host bridge overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 3.10.0-514.2.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.64 GiB
Name: k8master
ID: GAED:DBFS:FAUG:JEOP:ELGZ:MEBV:SU76:YAJ6:QRK4:EYDV:YVF3:XNB5
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8

**Comment by gaurav-dalvi** [comment]
@shaitmm : Issue is there because of docker 1.12.x . We had no issues with docker 1.11. If you want to try contiv you can try that. If you are strict on Docker 1.12 then I would suggest to wait for couple of days until we resolve above issue.

Docker 1.11 with Docker swarm + Contiv Demo : https://github.com/contiv/demo/tree/master/net

**Comment by shaitmm** [comment]
@gaurav-dalvi hi bro, this is the one i use which install docker 1.10.3 as per my comment

**Comment by gaurav-dalvi** [comment]
it installs docker 1.11.1

https://github.com/contiv/demo/blob/master/net/net_demo_installer#L197

**Comment by shaitmm** [comment]
@gaurav-dalvi  im Installing Contiv Network with Docker Swarm and stucked at “TASK [etcd: copy the etcd start/stop script] node 3 failed AnsibleUndefinedVarable：’Dict object' has no attribute u' ansible_ens19'". Seems after ansible get iptalbes configured, ens192 on every node are not accessable anymore.

pls find env info as below.
[root@localhost net]# cat cfg.yml 
CONNECTION_INFO:
 10.74.82.253:
  control: ens192
  data: ens160

 10.74.82.246:
  control: ens192
  data: ens160

 10.74.82.247:
  control: ens192
  data: ens160


[root@localhost net]# cat .gen/hosts.log 
node3 | SUCCESS => {
    "ansible_facts": {
        "ansible_distribution": "CentOS", 
        "ansible_distribution_major_version": "7", 
        "ansible_distribution_release": "Core", 
        "ansible_distribution_version": "7.3.1611"
    }, 
    "changed": false
}
node1 | UNREACHABLE! => {
    "changed": false, 
    "msg": "Failed to connect to the host via ssh: ssh: connect to host 10.74.82.246 port 22: Connection timed out\r\n", 
    "unreachable": true
}
node2 | UNREACHABLE! => {
    "changed": false, 
    "msg": "Failed to connect to the host via ssh: ssh: connect to host 10.74.82.247 port 22: Connection timed out\r\n", 
    "unreachable": true
}

[netplugin-node]
node1 ansible_ssh_host=10.74.82.246 contiv_network_mode=standalone control_interface=ens192 netplugin_if=ens160 fwd_mode=bridge
node2 ansible_ssh_host=10.74.82.247 contiv_network_mode=standalone control_interface=ens192 netplugin_if=ens160 fwd_mode=bridge
node3 ansible_ssh_host=10.74.82.253 contiv_network_mode=standalone control_interface=ens192 netplugin_if=ens160 fwd_mode=bridge

**Comment by shaitmm** [comment]
@gaurav-dalvi im Installing Contiv Network with Docker Swarm and stucked at “TASK [etcd: copy the etcd start/stop script] node 3 failed AnsibleUndefinedVarable：’Dict object' has no attribute u' ansible_ens192'". Seems after ansible get iptalbes configured, ens192 on every node are not accessable anymore.

pls find env info as below.
[root@localhost net]# cat cfg.yml
CONNECTION_INFO:
10.74.82.253:
control: ens192
data: ens160

10.74.82.246:
control: ens192
data: ens160

10.74.82.247:
control: ens192
data: ens160

[root@localhost net]# cat .gen/hosts.log
node3 | SUCCESS => {
"ansible_facts": {
"ansible_distribution": "CentOS",
"ansible_distribution_major_version": "7",
"ansible_distribution_release": "Core",
"ansible_distribution_version": "7.3.1611"
},
"changed": false
}
node1 | UNREACHABLE! => {
"changed": false,
"msg": "Failed to connect to the host via ssh: ssh: connect to host 10.74.82.246 port 22: Connection timed out\r\n",
"unreachable": true
}
node2 | UNREACHABLE! => {
"changed": false,
"msg": "Failed to connect to the host via ssh: ssh: connect to host 10.74.82.247 port 22: Connection timed out\r\n",
"unreachable": true
}

[netplugin-node]
node1 ansible_ssh_host=10.74.82.246 contiv_network_mode=standalone control_interface=ens192 netplugin_if=ens160 fwd_mode=bridge
node2 ansible_ssh_host=10.74.82.247 contiv_network_mode=standalone control_interface=ens192 netplugin_if=ens160 fwd_mode=bridge
node3 ansible_ssh_host=10.74.82.253 contiv_network_mode=standalone control_interface=ens192 netplugin_if=ens160 fwd_mode=bridge

**Comment by shaitmm** [comment]
@gaurav-dalvi im Installing Contiv Network with Docker Swarm and stucked at “TASK [etcd: copy the etcd start/stop script] node 3 failed AnsibleUndefinedVarable：’Dict object' has no attribute u' ansible_ens192'". Seems after ansible get iptalbes configured, ens192 on every node are not accessable anymore.

pls find env info as below.
[root@localhost net]# cat cfg.yml
CONNECTION_INFO:
10.74.82.253:
control: ens192
data: ens160

10.74.82.246:
control: ens192
data: ens160

10.74.82.247:
control: ens192
data: ens160

[root@localhost net]# cat .gen/hosts.log
node3 | SUCCESS => {
"ansible_facts": {
"ansible_distribution": "CentOS",
"ansible_distribution_major_version": "7",
"ansible_distribution_release": "Core",
"ansible_distribution_version": "7.3.1611"
},
"changed": false
}
node1 | UNREACHABLE! => {
"changed": false,
"msg": "Failed to connect to the host via ssh: ssh: connect to host 10.74.82.246 port 22: Connection timed out\r\n",
"unreachable": true
}
node2 | UNREACHABLE! => {
"changed": false,
"msg": "Failed to connect to the host via ssh: ssh: connect to host 10.74.82.247 port 22: Connection timed out\r\n",
"unreachable": true
}

[netplugin-node]
node1 ansible_ssh_host=10.74.82.246 contiv_network_mode=standalone control_interface=ens192 netplugin_if=ens160 fwd_mode=bridge
node2 ansible_ssh_host=10.74.82.247 contiv_network_mode=standalone control_interface=ens192 netplugin_if=ens160 fwd_mode=bridge
node3 ansible_ssh_host=10.74.82.253 contiv_network_mode=standalone control_interface=ens192 netplugin_if=ens160 fwd_mode=bridge

**Comment by gaurav-dalvi** [comment]
@shaitmm : I know this error. If you give incorrect interface name / or un reachable interface name in cfg.yml, you get that error. Please kindly check your cfg.yml again. 
I hope you are using ansible 2.2.0.0

**Comment by shaitmm** [comment]
@gaurav-dalvi 
control: ens160
data: ens192
initially i use above cfg in cfg.yml which caused the same FW issue.
But i tried to swap the interface of control and data as above this time, i finally got swarm and contiv setup. 
BTW, pls let me know once the issue i originally reported is resolve. i will give another try on k8s.

my ansible is the one you mentioned.
Name        : ansible
Arch        : noarch
Version     : 2.2.0.0
Release     : 4.el7
Size        : 21 M

**Comment by gaurav-dalvi** [comment]
Yes. we will have PR today or tomorrow and I will reply here once that is merged !
Thanks for your patience !

**Comment by shaitmm** [comment]
@gaurav-dalvi hi bro, hopefully the issue is resolved on k8s, could you pls show me the changes of the new release.

**Comment by gaurav-dalvi** [comment]
@shaitmm : Can you please check with docker swarm + contiv ? 

Do git clone 
https://github.com/contiv/netplugin/

make demo

make ssh

export DOCKER_HOST=tcp://192.168.2.10:3375

You are all se !

**Comment by shaitmm** [comment]
@gaurav-dalvi  
its working on swarm but not on k8s yet. i thought the issue on k8s could be resolved after the release is ready as per-talk.

**Comment by gaurav-dalvi** [comment]
ok. I will let @gkvijay to comment on this one.
@gkvijay : We are not seeing network, in case of K8s setup.

**Comment by shaitmm** [comment]
@gkvijay  could you pls guide me through contiv setup with K8s

**Comment by shaitmm** [comment]
@gkvijay  hi bro, your input will be highly appreciated.

**Comment by gkvijay** [comment]
@shaitmm, in k8s environment the docker commands will not show the contiv networks. We dont use docker daemon to create the networks. k8s handles it.
